registration-processor/pre-processor/registration-processor-packet-validator-stage/src/main/java/io/mosip/registration/processor/stages/utils/BiometricsXSDValidator.java

```diff
@@ -0,0 +1,108 @@
+package io.mosip.registration.processor.stages.utils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.io.IOUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import io.mosip.kernel.biometrics.constant.BiometricType;
+import io.mosip.kernel.biometrics.entities.BiometricRecord;
+import io.mosip.kernel.cbeffutil.container.impl.CbeffContainerImpl;
+import io.mosip.kernel.core.cbeffutil.common.CbeffValidator;
+import io.mosip.kernel.core.cbeffutil.entity.BDBInfo;
+import io.mosip.kernel.core.cbeffutil.entity.BIR;
+import io.mosip.kernel.core.cbeffutil.entity.BIRInfo;
+import io.mosip.kernel.core.cbeffutil.entity.BIRVersion;
+import io.mosip.kernel.core.cbeffutil.exception.CbeffException;
+import io.mosip.kernel.core.cbeffutil.jaxbclasses.BIRType;
+import io.mosip.kernel.core.cbeffutil.jaxbclasses.ProcessedLevelType;
+import io.mosip.kernel.core.cbeffutil.jaxbclasses.PurposeType;
+import io.mosip.kernel.core.cbeffutil.jaxbclasses.QualityType;
+import io.mosip.kernel.core.cbeffutil.jaxbclasses.RegistryIDType;
+import io.mosip.kernel.core.cbeffutil.jaxbclasses.SingleType;
+import io.mosip.registration.processor.core.packet.dto.packetvalidator.PacketValidationDto;
+import io.mosip.registration.processor.core.status.util.StatusUtil;
+@Component
+public class BiometricsXSDValidator {
+
+    @Value("${mosip.kernel.xsdstorage-uri}")
+    private String configServerFileStorageURL;
+    
+    @Value("${mosip.kernel.xsdfile}")
+    private String schemaFileName;
+    
+    public boolean validateXSD(BiometricRecord biometricRecord, PacketValidationDto packetValidationDto ) throws Exception  {
+    	try (InputStream xsd = new URL(configServerFileStorageURL + schemaFileName).openStream()) {
+            CbeffContainerImpl cbeffContainer = new CbeffContainerImpl();
+            List<BIR> birList = new ArrayList<>();
+            biometricRecord.getSegments().forEach(s -> birList.add(convertToBIR(s)));
+            BIRType bir = cbeffContainer.createBIRType(birList);
+            try{
+            CbeffValidator.createXMLBytes(bir, IOUtils.toByteArray(xsd));//validates XSD 
+            }catch(CbeffException e) {
+            	packetValidationDto.setPacketValidaionFailureMessage(StatusUtil.XSD_VALIDATION_EXCEPTION.getMessage()+e.getMessage());

```

Comment:
can't this be added to the packet validator impl layer like in other validators?

Reply:
The message is coming in exception . Since the return statement is in catch clause another exception can not be thrown there. If I move it to packet validator impl layer the message will be generic .